tests/test_flows.py

```diff
@@ -225,6 +295,82 @@ def test_initial_ap_config(self):
+
+        # Wait a bit for the renew to complete
+        time.sleep(3)
+
+        self.check_log(self.repeater1, "agent_wlan0", r"Received credentials for ssid: Multi-AP-24G-1 .* bss_type: 2")
+        self.check_log(self.repeater1, "agent_wlan0", r"Received credentials for ssid: Multi-AP-24G-2 .* bss_type: 1")
+        self.check_log(self.repeater1, "agent_wlan2", r".* tear down radio")
+
+        bssid1 = self.repeater1_ucc.dev_get_parameter('macaddr',
+                                                      ruid = '0x' + self.mac_repeater1_wlan0.replace(':', ''),
+                                                      ssid = 'Multi-AP-24G-1')
+        if not bssid1:
+            self.fail("repeater1 didn't configure Multi-AP-24G-1")
+
+    def test_ap_config_bss_tear_down(self):
+        # Configure the controller and send renew
+        self.gateway_ucc.cmd_reply("DEV_RESET_DEFAULT")
+        self.gateway_ucc.cmd_reply(
+            "DEV_SET_CONFIG,bss_info1,"
+                "{self.mac_repeater1} 8x Multi-AP-24G-3 0x0020 0x0008 maprocks1 0 1".format(self = self))
+        self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x000A,
+            tlv(0x01, 0x0006, "{" + self.mac_gateway + "}"),
+            tlv(0x0F, 0x0001, "{0x00}"),
+            tlv(0x10, 0x0001, "{0x00}"))
+
+        # Wait a bit for the renew to complete
+        time.sleep(3)
+
+        self.check_log(self.repeater1, "agent_wlan0", r"Received credentials for ssid: Multi-AP-24G-3 .* bss_type: 2")
+        self.check_log(self.repeater1, "agent_wlan2", r".* tear down radio")
+        conn_map = self.get_conn_map()
+        repeater1 = conn_map[self.mac_repeater1]
+        repeater1_wlan0 = repeater1.radios[self.mac_repeater1_wlan0]
+        for vap in repeater1_wlan0.vaps.values():
+            if vap.ssid not in (b'Multi-AP-24G-3', b'N/A'):
+                self.fail('Wrong SSID: {vap.ssid} instead of Multi-AP-24G-3'.format(vap = vap))
+        repeater1_wlan2 = repeater1.radios[self.mac_repeater1_wlan2]
+        for vap in repeater1_wlan2.vaps.values():
+            if vap.ssid != b'N/A':
+                self.fail('Wrong SSID: {vap.ssid} instead torn down'.format(vap = vap))
+
+        # SSIDs have been removed for the CTT Agent1's front radio
+        self.gateway_ucc.cmd_reply(
+            "DEV_SET_CONFIG,bss_info1,{self.mac_repeater1} 8x".format(self = self))
+        # Send renew message
+        self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x000A,
+            tlv(0x01, 0x0006, "{" + self.mac_gateway + "}"),
+            tlv(0x0F, 0x0001, "{0x00}"),
+            tlv(0x10, 0x0001, "{0x00}"))
+
+        time.sleep(3)
+        self.check_log(self.repeater1, "agent_wlan0", r".* tear down radio")
+        conn_map = self.get_conn_map()
+        repeater1 = conn_map[self.mac_repeater1]
+        repeater1_wlan0 = repeater1.radios[self.mac_repeater1_wlan0]
+        for vap in repeater1_wlan0.vaps.values():
+            if vap.ssid != b'N/A':
```

Comment:
Nitpick: `SSID`s should probably be strings, not `bytes`, since they can contain unicode

Reply:
SSIDs can contain any bytes, not limited to UTF-8. So it should most definitely be `bytes`.